### DISCUSSION TITLE
How to try several upstream proxies in case of failure?

### DISCUSSION THREAD

**Discussion by ZzZombo** [mitmproxy/mitmproxy#6007] [Q&A]
The things I don't know how to handle: how do I know if MITMProxy encountered an error talking to the current upstream proxy? Only such errors should result in trying a different proxy as, say, connection timeout or refused, not something like the target site being down and reported as such by the upstream proxy. I'm not also sure at all what addon error handling is supposed to be at all, because I need to retry on the same client connection, invisibly to the client.

**Comment by ZzZombo** [comment]
Could `ctx.master.load_flow(f)` be used to try a different proxy server in an `error` handler I wonder where `f` is the modified original flow with a new proxy set.

**Comment by ZzZombo** [comment]
Sorry, why it appears that all my questions are ignored? Given this IS the official support channel, I have no choice but to abide:

> If you have questions on how to use mitmproxy, ask them on the discussions page!

Please, I can see there is quite some activity going on, could some developer lend me a hand?

**Reply by mhils** [reply]
> Sorry, why it appears that all my questions are ignored?


Looking at the [issue tracker](https://github.com/mitmproxy/mitmproxy/issues?q=author%3AZzZombo), I don't think it's fair to claim that you are being ignored. I'm happy to hook you up with a professional support contract and well-defined SLAs though. 😉 

Instead of complaining about the lack of replies, you could have searched the tracker for similar items. Your question is already answered in https://github.com/mitmproxy/mitmproxy/discussions/5023: That's nothing we support very well. Please keep in mind that we are all volunteers.
